### PR TITLE
[CI] explicitly select if mainnet should be built

### PR DESCRIPTION
**Pull request by dkijania**
After we metg

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!help-merge-me

**Comment by dkijania** [comment]
Hello, @dkijania 2 new  created to help you port this change to master. Please follow steps to resolve conflict for each mainline branch. PRs will be created automatically after you push merge to porting branches

  
## develop 

```
git fetch && git checkout port/_ci/build_mainnet_and_devnet_in_ci_to_develop && git merge ci/build_mainnet_and_devnet_in_ci
```

Resolve conflict and then 

```
git push
```  
    
  
## compatible 

```
git fetch && git checkout port/_ci/build_mainnet_and_devnet_in_ci_to_compatible && git merge ci/build_mainnet_and_devnet_in_ci
```

Resolve conflict and then 

```
git push
```

**Comment by dkijania** [comment]
Porting branch #15800 to resolve conflict with develop created. Please merge it in order to solve conflict between head of this PR and develop

**Comment by dkijania** [comment]
Porting branch #15801 to resolve conflict with compatible created. Please merge it in order to solve conflict between head of this PR and compatible

**Comment by dkijania** [comment]
!ci-build-me

File: buildkite/src/Command/MinaArtifact.dhall [hunk 1]
```diff
@@ -206,118 +205,156 @@ let hardforkPipeline : DebianVersions.DebVersion -> Pipeline.Config.Type =
...
+        \(spec: MinaBuildSpec.Type) ->
+
+       let docker_steps = List/map
+              Artifacts.Type
+              (List DockerImage.ReleaseSpec.Type)
+              (\(artifact: Artifacts.Type) -> docker_step artifact spec.debVersion spec.networks spec.profile)
+              spec.artifacts
+      
+      
+      let flatten_docker_steps =
```
Comment (georgeee) [inline buildkite/src/Command/MinaArtifact.dhall:327]:
```suggestion
      let flattened_docker_steps = 
```

File: buildkite/src/Command/MinaArtifact.dhall [hunk 2]
```diff
@@ -206,118 +205,156 @@ let hardforkPipeline : DebianVersions.DebVersion -> Pipeline.Config.Type =
...
+              (List DockerImage.ReleaseSpec.Type)
+              (\(x : List DockerImage.ReleaseSpec.Type) → \(y : List DockerImage.ReleaseSpec.Type) →  x # y)
+              ([] : List DockerImage.ReleaseSpec.Type)
+      in 
+
+      (List/map
+            DockerImage.ReleaseSpec.Type
+            Command.Type
+            (\(s: DockerImage.ReleaseSpec.Type) -> DockerImage.generateStep s)
+            flatten_docker_steps)
```
Comment (georgeee) [inline buildkite/src/Command/MinaArtifact.dhall:340]:
```suggestion
            flattened_docker_steps)
```

File: scripts/rebuild-deb.sh [hunk 3]
```diff
@@ -16,6 +16,8 @@ if [ $# -eq 0 ]
     build_archive_migration_deb
     build_batch_txn_deb
     build_daemon_deb
```
Comment (georgeee) [inline scripts/rebuild-deb.sh:18]:
```suggestion
    build_daemon_berkeley_deb
```

**Review by georgeee — CHANGES_REQUESTED** [review]
PR seems to contain a couple of bugs that suggest that testing performed was limited at the point. Overall, PR seems to be implementing what it suggests to implement, however our Dhall code becomes even messier than it was before.

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by SanabriaRusso** [comment]
I just wanted to run a nightly to make sure: https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2216#01909261-e7a4-4c43-a6af-72a6ac8adeff

All build tests passed OK

**Review by SanabriaRusso — APPROVED** [review]

**Review by svv232 — APPROVED** [review]

**Pull request merged**